File: restdocWithSwagger/src/main/java/com/example/restdocwithswagger/SecurityConfig.java
package com.example.restdocwithswagger;

import org.springframework.context.annotation.Bean;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.web.SecurityFilterChain;

@EnableWebSecurity
public class SecurityConfig {

    private static final String[] PERMIT_URL_ARRAY = {
            /* swagger v2 */
            "/v2/api-docs",
            "/swagger-resources",
            "/swagger-resources/**",
            "/configuration/ui",
            "/configuration/security",
            "/index.html",
            "/webjars/**",
            /* swagger v3 */
            "/v3/api-docs/**",
            "/swagger-ui/**"
    };

    @Bean
    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
            return http.csrf().disable()
                    .authorizeRequests()
                    .antMatchers(PERMIT_URL_ARRAY).permitAll()
//                    .mvcMatchers("/spring3/api/lecture/**").permitAll()
                    .anyRequest().permitAll()
                .and().build();
    }
}

File: kotilWebSecurity/src/main/kotlin/kr/pe/deity/kotilwebsecurity/SecurityConfig.kt
package kr.pe.deity.kotilwebsecurity

import org.springframework.context.annotation.Bean
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.web.SecurityFilterChain

@EnableWebSecurity
class SecurityConfig{

    @Bean
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain{
        return http.csrf().disable()
            .authorizeRequests()
            .antMatchers("/v2/api-docs",
                "/swagger-resources",
                "/swagger-resources/**",
                "/configuration/ui",
                "/configuration/security",
                "/index.html",
                "/webjars/**",
                /* swagger v3 */
                "/v3/api-docs/**",
                "/swagger-ui/**").permitAll()
//                    .mvcMatchers("/spring3/api/lecture/**").permitAll()
            .anyRequest().permitAll()
            .and().build();
    }
}